### PR TITLE
Make sure we can actually look up the values for keys to containers in the variable view

### PR DESCRIPTION
**Pull request by mvanderkamp**
Defining keys() but not __getitem__ is a pretty big code smell, but that
doesn't mean we should crash.

### CODE REVIEW

File: pudb/var_view.py [hunk 1]
```diff
@@ -432,7 +432,7 @@ def walk_value(self, parent, label, value, id_path=None, attr_prefix=None):
                     else:
                         key_it = xrange(len_value)
 
-            if key_it is not None:
+            if key_it is not None and hasattr(value, '__getitem__'):
```
Comment (inducer) [inline pudb/var_view.py:435]: Wouldn't it be more robust/Pythonic to handle failure of `value[key]`?

Comment (mvanderkamp) [reply pudb/var_view.py:435]: I'll take a look at this. I've also encountered some other similar instances where we could be a bit more robust about catching errors when trying to figure out how to walk a value.

**Comment by mvanderkamp** [comment]
I think that should work. I was thinking of testing the `walk_value` method to make sure that all these various edge cases are handled and continue to behave reasonably, but I might do that separately unless you think that it would be a good thing to add here.

File: pudb/var_view.py [hunk 2]
```diff
@@ -417,15 +417,15 @@ def walk_value(self, parent, label, value, id_path=None, attr_prefix=None):
             if key_it is None:
                 try:
                     len_value = len(value)
-                except TypeError:
+                except (AttributeError, TypeError):
```
Comment (inducer) [inline pudb/var_view.py:420]: Under what circumstances does `AttributeError` happen here?
```py
>>> class A:
...    pass
... 
>>> a = A()
>>> len(a)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: object of type 'A' has no len()
```

**Review by inducer — APPROVED** [review]
Thanks, LGTM! One question, but good to go at any rate.

**Comment by inducer** [comment]
>  I was thinking of adding unit tests for the walk_value method to make sure that all these various edge cases are handled and continue to behave reasonably, but I might do that separately unless you think that it would be a good thing to add here.

I like this idea. Thanks for offering to add that.

File: pudb/var_view.py (hunk 2 above)
Comment (mvanderkamp) [reply pudb/var_view.py:420]: You know, I'm not sure. I added that at some point during a marathon debugging session at work but forgot to copy down the traceback, if there even was one. I don't mind removing it.

Comment (mvanderkamp) [reply pudb/var_view.py:420]: (it's possible it was even some sort of internal bug in a `__len__` implementation)

Comment (inducer) [reply pudb/var_view.py:420]: Thanks for answering! I don't think it hurts.
